packages/components/src/Link/Link.stories.tsx

```diff
@@ -0,0 +1,242 @@
+  ...defaultArgs,
+  children: "Link Button",
+  variant: "link",
+};
+
+export const withDataTestId = Template.bind(null);
+withDataTestId.args = {
+  ...defaultArgs,
+  children: "Link with data-testid",
+  color: "alert",
+  "data-testid": "fake-test-id",
+};
+
+export const linkWithIcon = Template.bind(null);
+linkWithIcon.args = {
+  ...defaultArgs,
+  children: (
+    <>
+      Link with icon <CheckCircleRoundedIcon purpose="decorative" />
+    </>
+  ),
+  color: "success",
+  variant: "link",
+};
+
+export const outlineWithIcon = Template.bind(null);
+outlineWithIcon.args = {
+  ...defaultArgs,
+  children: (
+    <>
+      Outline with icon <CheckCircleRoundedIcon purpose="decorative" />
+    </>
+  ),
+  color: "warning",
+  variant: "outline",
+};
+
+export const withFakeClassName = Template.bind(null);
+withFakeClassName.args = {
+  ...defaultArgs,
+  children: "With fake className",
+  color: "warning",
+  variant: "outline",
+  className: "fake-className",
+};
+
+// Show grids with all variants
+
+const gridParameters = {
+  axe: {
+    skip: true,
+  },
+  snapshot: {
+    skip: true,
+  },
+};
+
+const renderSize = (
+  size: LinkProps["size"],
+  textColor: "white" | "neutral",
+  children: React.ReactNode,
+) =>
+  variants.map((variant) => {
+    const colors = variant === "flat" ? flatColors : allColors;
+
+    return (
+      <React.Fragment key={variant}>
+        <Heading size="h2" color={textColor}>
+          {variant} - {size}
+        </Heading>
+        <table className={styles.variant}>
+          <tbody>
+            {states.map((state) => (
+              <tr key={state}>
+                <th scope="row">
+                  <Text size="body" color={textColor}>
+                    {state}
+                  </Text>
+                </th>
+                {colors.map((color) => (
+                  <td key={color} className={styles.color}>
+                    {/* To pass the "state" prop (only used for demonstration in storybook),
+                    we must use Clickable instead of Link */}
+                    <Clickable
+                      as={"a"}
+                      size={size}
+                      color={color}
+                      variant={variant}
+                      state={state}
+                      href=""
+                    >
+                      {children}
+                    </Clickable>
+                  </td>
+                ))}
+              </tr>
+            ))}
+          </tbody>
+        </table>
+      </React.Fragment>
+    );
+  });
+
+export const allVariants = () => (
+  <ul>
+    {sizes.map((size) => (
+      <li key={size}>{renderSize(size, "neutral", "Link")}</li>
+    ))}
+  </ul>
+);
+
+allVariants.parameters = gridParameters;
+
+export const mediumVariantsOnDarkBackground = () =>
+  renderSize("medium", "white", "Link");
+
+mediumVariantsOnDarkBackground.parameters = {
+  ...gridParameters,
+  backgrounds: {
+    default: "dark",
+  },
+};
```

Comment:
Think you could replace this with

```
import * as ButtonStoryFile from "../Button/button.stories";

export const allVariants = ButtonStoryFile.allVariants;
allVariants.parameters = gridParameters;

export const mediumVariantsOnDarkBackground =
  ButtonStoryFile.mediumVariantsOnDarkBackground;
mediumVariantsOnDarkBackground.parameters = {
  ...gridParameters,
  backgrounds: {
    default: "dark",
  },
};
```

Reply:
The only thing is the `Button` component is using a `button` tag in the grid and it would be really nice to have this one use an anchor tag for more thorough testing.

Reply:
Oh and the other difference is the `disabled` row isn't applicable for this component since you can't disable links.

Reply:
oh, right 👍